nixos/modules/security/doas.nix

```
@@ -237,6 +243,10 @@ in
   ###### implementation
 
   config = mkIf cfg.enable {
+    assertions = singleton {
+      assertion = !(cfg.sudoShim.enable && config.security.sudo.enable);
+      message = "security.doas.sudoShim.enable conflicts with security.sudo.enable";
+    };
```

Personally I would just use this instead of singleton here. Easier to read for people who aren't as familiar with nixpkgs.lib and reduces the diff in case options are added in the future. (Also even without those I don't think using singleton is really gaining anything here, it's mainly nice as part of list concat together with `optional` and so on)

```suggestion
    assertions = [
      {
        assertion = !(cfg.sudoShim.enable && config.security.sudo.enable);
        message = "security.doas.sudoShim.enable conflicts with security.sudo.enable";
      }
    ];
```